atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockService.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+
+/**
+ * Provides endpoints for acquiring and releasing the Deletion Lock. This is intended to be used by backups and sweep,
+ * to ensure that only one of these operations is deleting data at once.
+ */
+@Path("/persistent-lock")
+public interface PersistentLockService {
+    /**
+     * Attempt to acquire the lock.
+     * Call this method before performing any destructive operations.
+     * @param reason the reason for the lock, for logging purposes (e.g. "sweep")
+     * @return a {@link LockEntry} on success. The LockEntry will contain the given reason, and a unique ID. It is
+     *   essential that you retain a reference to this lock, as you will need it in order to release the lock.
+     */
+    @POST // This has to be POST because we can't allow caching.
+    @Path("acquire")
+    @Produces(MediaType.APPLICATION_JSON)
+    LockEntry acquireLock(@QueryParam("reason") String reason);
```

Comment:
After discussion we have decided that this api is appropriate as it can be extended to allow other locks, by allowing the user to supply a lock name as well.  This would be an additive change.  My only remaining concern her is should this endpoint reference that this is the backup lock?  `acquire-backup-lock` so that when made more generic it doesn't contain an underspecified `acquire` lock endpoint.